spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -244,6 +244,10 @@ private boolean requiresRewrite(Filter filter, Schema schema, Set<Integer> ident
 
   @Override
   public void deleteWhere(Filter[] filters) {
+    Preconditions.checkArgument(
+        snapshotId == null,
+        "Cannot delete from table at a specific snapshot: %s", snapshotId);
+
```

Comment:
Can you please remove this same check from `canDeleteWhere` above? That way, the check is present in only one place, the place where it is actually called.
IIUC, Spark 3.0 doesn't actually call `SparkTable#canDeleteWhere` (Spark 3.1 does); we have `canDeleteWhere` in the code here because it was added when we had a single code base for both Spark 3.0 and 3.1.

Reply:
I don't think that the implementation of `canDeleteWhere` matters much for Spark 3.0. We could simply remove the function since it is unused, but it doesn't seem worth letting the code drift either to remove it or to modify it to have this check in just one place. So I guess I disagree with this suggestion. Let's just leave it as is.